mlx/coverity_services.py

```diff
@@ -1,614 +1,396 @@
+        Args:
+            column_name (str): The column name in lowercase
+            filter_values (list[str]): The list of valid values to filter on
+            matcher_type (str): The type of the matcher (nameMatcher, idMatcher or keyMatcher)
 
-        snapshot_scope.showOutdatedStreams = False
-        snapshot_scope.compareOutdatedStreams = False
+        Returns:
+            dict: New filter for API queries
+        """
+        matchers = []
+        for filter_ in filter_values:
+            matcher = {"type": matcher_type}
+            if matcher_type == "nameMatcher":
+                matcher["name"] = filter_
+                matcher["class"] = "Component"
+                assert column_name == "component"
+            elif matcher_type == "idMatcher":
+                matcher["id"] = filter_
+            else:
+                matcher["key"] = filter_
+            matchers.append(matcher)
+        return {
+            "columnKey": self.columns[column_name],
+            "matchMode": "oneOrMoreMatch",
+            "matchers": matchers
+        }
+
+    def get_defects(self, stream, filters, column_names):
+        """Gets a list of defects for given stream, filters and column names.
+        If a column name does not match the name of the `columns` property, the column can not be obtained because
+        it need the correct corresponding column key.
+        Column key `cid` is always obtained to use later in other functions.
 
-        snapshot_scope.showSelector = 'last()'
-        snapshot_scope.compareSelector = 'last()'
+        Args:
+            stream (str): Name of the stream to query
+            filters (dict): Dictionary with attribute names as keys and CSV lists of attribute values to query as values
+            column_names (list[str]): The column names
 
-        logging.info('Running Coverity query...')
-        return self.client.service.getMergedDefectsForSnapshotScope(project_id, filter_spec,
-                                                                    page_spec, snapshot_scope)
+        Returns:
+            dict: The content of the request. This has a structure like:
+                {
+                    "offset": 0,
+                    "totalRows": 2720,
+                    "columns": [list of column keys]
+                    "rows": list of [list of dictionaries {"key": <key>, "value": <value>}]
+                }
+        """
+        logging.info("Querying Coverity for defects in stream [%s] ...", stream)
+        query_filters = [
+            {
+                "columnKey": "streams",
+                "matchMode": "oneOrMoreMatch",
+                "matchers": [
+                    {
+                        "class": "Stream",
+                        "name": stream,
+                        "type": "nameMatcher"
+                    }
+                ]
+            }
+        ]
+
+        Filter = namedtuple("Filter", "name matcher_type list allow_regex", defaults=[None, False])
```

Comment:
```suggestion
        Filter = namedtuple("Filter", "name matcher_type list allow_regex", defaults=[[], False])
```
`None` would result in an error when combined with `allow_regex=True`.

`for element in valid_attributes:` https://github.com/melexis/sphinx-coverity-extension/pull/74/files?file-filters%5B%5D=.py&show-viewed-files=true&show-deleted-files=false#diff-68fcdda1ac3f8f1b5537915662685a35e40be83d207fd57e42da4008a2ad4e63R232

Reply:
Resolved in 1ab9fa00e69b63cd8465272bf40285fac06d8144